<launch>
  <arg name="manager" default="manager" />
  <arg name="standalone" default="false" />
  <arg name="image" />

  <arg if="$(arg standalone)" name="nodelet_args" value="standalone feature_extraction/FeatureExtractor" />
  <arg unless="$(arg standalone)" name="nodelet_args" value="load feature_extraction/FeatureExtractor $(arg manager)" />

  <node name="feature_extractor" pkg="nodelet" type="nodelet" args="$(arg nodelet_args)" output="screen" >
    <remap from="image" to="$(arg image)" />
    <param name="max_num_key_points" value="5000" />
    <param name="feature_extractor" value="SmartSURF" />
    <param name="show_image" value="true" />
  </node>

</launch>
